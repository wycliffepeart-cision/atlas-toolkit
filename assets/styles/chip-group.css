@layer atlas {
  atlas-chip-group {
    --overflow-background: var(--atlas-color-white);
    --overflow-border-color: var(--atlas-color-primary-n200);
    --overflow-color: var(--atlas-color-primary-n400);
  }

  atlas-chip-group {
    display: flex;
    flex-wrap: wrap;
    padding-block: 1em;
    row-gap: 1em;
    column-gap: 0.7em;
  }

  atlas-chip-group[category] > atlas-chip[slot='suffix'] {
    border-color: var(--overflow-border-color);
    background-color: var(--overflow-background);
    color: var(--overflow-color);
  }

  atlas-chip-group:not([category]) > atlas-chip[slot='suffix'] {
    border-color: var(--overflow-border-color);
    background-color: var(--overflow-background);
    color: var(--overflow-color);
  }

  atlas-chip-group[category='outlets'] {
    --atlas-chip-category-border-color: var(--atlas-chip-outlets-border-color);
    --atlas-chip-category-background-color: var(--atlas-chip-outlets-background-color);
    --atlas-chip-category-color: var(--atlas-chip-outlets-color);
    --atlas-chip-category-hover-background-color: var(--atlas-chip-outlets-color);
    --atlas-chip-category-hover-color: var(--atlas-chip-outlets-hover-color);
    --atlas-chip-category-active-color: var(--atlas-chip-outlets-color);
    --atlas-chip-category-active-shadow-color: var(--atlas-chip-outlets-active-shadow-color);
    --atlas-chip-category-active-background-color: var(--atlas-chip-outlets-active-background-color);
  }

  atlas-chip-group[category='hashtags'] {
    --atlas-chip-category-border-color: var(--atlas-chip-hashtags-border-color);
    --atlas-chip-category-background-color: var(--atlas-chip-hashtags-background-color);
  }

  atlas-chip-group[category='topics'] {
    --atlas-chip-category-border-color: var(--atlas-chip-topics-border-color);
    --atlas-chip-category-background-color: var(--atlas-chip-topics-background-color);
    --atlas-chip-category-color: var(--atlas-chip-topics-color);
    --atlas-chip-category-hover-background-color: var(--atlas-chip-topics-color);
    --atlas-chip-category-hover-color: var(--atlas-chip-topics-hover-color);
    --atlas-chip-category-active-color: var(--atlas-chip-topics-color);
    --atlas-chip-category-active-shadow-color: var(--atlas-chip-topics-active-shadow-color);
    --atlas-chip-category-active-background-color: var(--atlas-chip-topics-active-background-color);
  }

  atlas-chip-group[category] > atlas-chip {
    border: 1px solid;
    border-color: var(--atlas-chip-category-border-color);
    background-color: var(--atlas-chip-category-background-color);
    color: var(--atlas-chip-category-color);
  }

  atlas-chip-group[category] atlas-chip[type='filter'] {
    color: var(--atlas-chip-type-filter-color);
    background-color: var(--atlas-chip-type-filter-background-color);
    border-color: var(--atlas-chip-type-filter-border-color);
  }

  atlas-chip-group[category] > atlas-chip[highlighted] {
    --atlas-chip-category-background-color: var(--atlas-chip-category-active-background-color);
    --atlas-chip-category-color: var(--atlas-chip-category-active-color);
  }
}
