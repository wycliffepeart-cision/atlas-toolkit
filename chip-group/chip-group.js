import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var ChipGroupTemplate = "<template>\n  <slot part=\"defaultGroup\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n</template>\n";

let ChipGroup = class ChipGroup extends HTMLElement {
};
ChipGroup = __decorate([
    Template(ChipGroupTemplate)
], ChipGroup);

export { ChipGroup };
