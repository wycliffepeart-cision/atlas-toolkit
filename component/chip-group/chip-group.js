import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var ChipGroupTemplate = "<template>\n  <slot part=\"defaultGroup\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n</template>\n";

let ChipGroup = class ChipGroup extends HTMLElement {
};
ChipGroup = __decorate([
    Template(ChipGroupTemplate)
], ChipGroup);
if (!customElements.get('atlas-chip-group'))
    customElements.define('atlas-chip-group', ChipGroup);

export { ChipGroup };
