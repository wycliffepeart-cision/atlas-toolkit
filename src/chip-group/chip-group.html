<template>
	<slot part="defaultGroup"></slot>
	<slot name="suffix" part="suffix"></slot>
</template>
