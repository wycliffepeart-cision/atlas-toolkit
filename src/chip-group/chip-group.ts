import ChipGroupTemplate from './chip-group.html';
import { Template } from '../decorators/template';

@Template(ChipGroupTemplate)
export class ChipGroup extends HTMLElement {}
