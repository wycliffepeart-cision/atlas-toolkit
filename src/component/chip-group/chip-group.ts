import ChipGroupTemplate from './chip-group.html';
import { Template } from '../../decorators/template';

@Template(ChipGroupTemplate)
export class ChipGroup extends HTMLElement {}

if (!customElements.get('atlas-chip-group')) customElements.define('atlas-chip-group', ChipGroup);
