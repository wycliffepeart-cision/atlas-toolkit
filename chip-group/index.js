import { ChipGroup } from './chip-group.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-chip-group'))
    customElements.define('atlas-chip-group', ChipGroup);
