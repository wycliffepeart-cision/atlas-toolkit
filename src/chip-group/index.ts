import { ChipGroup } from './chip-group'

if (!customElements.get('atlas-chip-group')) customElements.define('atlas-chip-group', ChipGroup)
