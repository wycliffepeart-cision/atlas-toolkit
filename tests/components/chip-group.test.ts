import '../../src/chip-group';

const ATLAS_CHIP_GROUP_TAG = 'atlas-chip-group';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;

const setupElement = async () => {
  await customElements.whenDefined(ATLAS_CHIP_GROUP_TAG);
  const chipGRoupElement = document.createElement(ATLAS_CHIP_GROUP_TAG);
  document.body.appendChild(chipGRoupElement);

  return chipGRoupElement;
};

afterEach(() => {
  const element = document.querySelector(ATLAS_CHIP_GROUP_TAG);
  if (element) {
    element.remove();
  }
});

describe('Atlas Chip Group', () => {
  it('is correctly appended in the DOM', async () => {
    const chipGRoupElement = await setupElement();

   ( expect(chipGRoupElement) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await setupElement();

    const selectedElement = document.querySelector(ATLAS_CHIP_GROUP_TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    const chipGRoupElement = await setupElement();

    (expect(chipGRoupElement) as any).toContainHTML('<atlas-chip-group></atlas-chip-group>');
    (expect(chipGRoupElement) as any).toBeEmptyDOMElement();
    (expect(getShadowDomContent(chipGRoupElement)) as any).toMatchSnapshot();
  });

});
