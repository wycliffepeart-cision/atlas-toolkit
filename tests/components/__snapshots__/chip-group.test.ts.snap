// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atlas Chip Group has the correct content when created 1`] = `
<slot part="defaultGroup">
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
`;
